.github/workflows/gradle.yml

```diff
@@ -0,0 +1,67 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'corretto'
+
+    # Configure Gradle for optimal use in GitHub Actions, including caching of downloaded dependencies.
+    # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582 # v4.0.0
```

Comment:
This commit seems to be for 4.0.0-rc1: https://github.com/gradle/actions/releases/tag/prerelease/v4
I don't see it on 4.0.0 (but maybe I'm not looking correctly): https://github.com/gradle/actions/compare/v4.0.0...main
There are more recent releases: https://github.com/gradle/actions/releases

Reply:
Official docs use `v4`: https://github.com/gradle/actions/blob/main/dependency-submission/README.md

Reply:
Seems like that specific commit comes just before v4.0.0 in v4 -- https://github.com/gradle/actions/commit/af1da67850ed9a4cedd57bfd976089dd991e2582. Not sure why the template uses that specific commit. In any case, I'll just use v4 rather than a specific commit.